packages/relay/test/utils/UserStateFactory.ts

```diff
@@ -4,26 +4,30 @@ import { UserState } from '@unirep/core'
 import { Identity } from '@semaphore-protocol/identity'
 import { ethers } from 'ethers'
 import { Prover } from '@unirep/circuits'
+import { UnirepSocialSynchronizer } from '../../src/synchornizer'
 
 export class UserStateFactory {
     db: DB
     provider: ethers.providers.JsonRpcProvider
     prover: Prover
     unirepAddress: string
     attesterId: string
+    synchronizer: UnirepSocialSynchronizer
```

Comment:
I see, thanks @vivianjeng, there is eventNames to map the handler, then adding unirepApp address is enough.

Reply:
Hi @vivianjeng sorry, after checking deeply, I am still confused why provide address / provder could fi this issue?

Reply:
Hi @vivianjeng after investigate, we found out the root cause is still the `Synchronizer`.
Since we create new  [`UnirepSocialSynchronizer`](https://github.com/social-tw/social-tw-website/blob/e083f3080c2652687faa95286e3ecee329d8d00f/packages/relay/src/index.ts#L42) and start to listen our new events, but our [userState](https://github.com/social-tw/social-tw-website/blob/e083f3080c2652687faa95286e3ecee329d8d00f/packages/frontend/src/contexts/User.tsx#L134) uses [the old Synchronizer](https://github.com/Unirep/Unirep/blob/b51a3a4363d3df2aca4c390849aeed3bb0d4ee47/packages/core/src/UserState.ts#L89) instead of `UnirepSocialSynchronizer`, so even call waitForSync, our new events will not be listened. 

Also we found the args [`db`](https://github.com/Unirep/Unirep/blob/b51a3a4363d3df2aca4c390849aeed3bb0d4ee47/packages/core/src/Synchronizer.ts#L540) in handleEvent method is type of TransactionDB, it's not Promise method, so we could remove the `await` before method.  

In sum, will change back to use `UnirepSocialSynchronizer` to fix this issue~